// Copyright 2019 The Cloud Robotics Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// + https://github.com/npclaudiu/grpc-cpp-docker
 
syntax = "proto3";
 
package helloworld;
 
// The greeting service definition.
service Greeter {
  // Sends a greeting.
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc Compute(ComputeRequest) returns (ComputeResponse) {}
}
 
// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}
 
// The response message containing the greetings.
message HelloReply {
  string message = 1;
}
 
enum Operator {
    ADD = 0;
    SUB = 1;
    MUL = 2;
    DIV = 3;
    POW = 4;
}
 
message ComputeRequest {
    Operator operator = 1;
    double lhs = 2;
    double rhs = 3;
}
 
message ComputeResponse {
    double result = 1;
}
